src/tools/rbd_mirror/ImageDeleter.cc

```diff
@@ -493,27 +486,52 @@ void ImageDeleter<I>::handle_retry_timer() {
       remove_images();
       m_async_op_tracker.finish_op();
     });
-  m_work_queue->queue(ctx, 0);
+  m_threads->work_queue->queue(ctx, 0);
 }
 
 template <typename I>
-void ImageDeleter<I>::DeleteInfo::notify(int r) {
-  if (on_delete) {
-    dout(20) << "executing image deletion handler r=" << r << dendl;
+void ImageDeleter<I>::handle_trash_image(const std::string& image_id,
+                                         const utime_t& deferment_end_time) {
+  Mutex::Locker timer_locker(m_threads->timer_lock);
+  Mutex::Locker locker(m_lock);
+
+  // TODO
```

Comment:
@dillaman not sure what this TODO means

Reply:
I address it in a later commit -- at this stage, ```ImageDeleter``` uses the ```global_image_id``` but it will be switched to ```local_image_id```

Reply:
I still see the comment in the final version. I suppose it should be removed?

Reply:
In that case -- definitely